src/api/functions.js

```diff
@@ -70,6 +70,46 @@ function _getCurrentCommunityContext(){
     return { __community: subdomain, __is_sandbox }
 }
 
+function set_cookie(cookies, key, value) {
+  cookies.set(key, value, { path: '/' });
+}
+
+function get_cookie(cookies, key) {
+  let cookie = cookies.get(key);
+  return cookie;
+}
+
+function log_device(cookies) {
+  // TODO: Get IP address and other info
+  let device = get_cookie(cookies, "device");
+  let body;
+  let apiFunction;
+  let response;
+  if (device === undefined) {
+    body = {};
+    apiFunction = 'create'
```

Comment:
since you combined the deviceprofile.create and deviceprofile.log in the API, why not just use .log?

Reply:
I haven't gotten to that yet. create and log used to have different functions. I made a change recently so I haven't gotten round to fixing that. It should be extremely easy.